### ISSUE TITLE
[ENHANCEMENT] sct_register_multimodal --> bsplinesyn --> add option for matrix size

### ISSUE THREAD

**Issue by tanguyduval** [neuropoly/spinalcordtoolbox#403]
currently default parameter for BsplineSyn is: 
--transform BSplineSyN[0.5,3,32]
Result (intensity range is -1e-6 to 1e-6):
![image](https://cloud.githubusercontent.com/assets/7785316/8661396/384ab5dc-2985-11e5-97b7-31d9ca7a9777.png)

However this parameter is critical because it defines the resolution/sharpness of the warping.
--transform BSplineSyN[0.5,1,3]
Result (intensity range is -1.29 to 0.21):
![image](https://cloud.githubusercontent.com/assets/7785316/8661402/446dd3bc-2985-11e5-8185-623f213ef6ec.png)

--transform BSplineSyN[0.5,3,3]
Result (intensity range is -1.92 to 0.67):
![image](https://cloud.githubusercontent.com/assets/7785316/8661455/973caf32-2985-11e5-971f-87a4e8af251d.png)

--transform BSplineSyN[0.5,1,1]
Result (intensity range is -2.81 to 2.43):
![image](https://cloud.githubusercontent.com/assets/7785316/8661466/acfd6b2c-2985-11e5-8ac7-edc6333aecd3.png)

if you increase:
--> second paramter: higher deformations
--> third parameter: global deformation

My choice for my images:
--transform BSplineSyN[0.5,1,3]

**Comment by tanguyduval** [comment]
Carefull because this parameter is hardcoded in both msct_register and sct_register_multimodal

**Comment by tanguyduval** [comment]
and m_sct_register_regularized

**Comment by tanguyduval** [comment]
search for:
ants_registration_params = {'rigid': '', 'affine': '', 'compositeaffine': '', 'similarity': '', 'translation': '',
                                'bspline': ',10', 'gaussiandisplacementfield': ',3,0',
                                'bsplinedisplacementfield': ',5,10', 'syn': ',3,0', 'bsplinesyn': ',1,3'}

**Comment by tanguymagnan** [comment]
It is now just hardcoded in sct_register_multimodal. Hardcoded as putting it in parameters may be too heavy (= too many parameters for sct_register_multimodal)

**Comment by benjamindeleener** [comment]
Indeed, this would still be a useful parameter, but a difficult one to master.

**Comment by joshuacwnewton** [comment]
Small note for repository maintenance purposes: `--transform` params are still hardcoded as of Nov 22, 2022:

https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/c57b43f715cb85dad87a33623c7caf0ed47543ca/spinalcordtoolbox/registration/core.py#L297-L300
